src/mlpack/core/cv/metrics/roc_auc_score_impl.hpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * @file core/cv/metrics/roc_auc_score_impl.hpp
+ * @author Sri Madhan M
+ *
+ * Implementation of the area under Receiver Operating Characteristic curve
+ * (ROC-AUC) score.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_METRICS_ROCAUCSCORE_IMPL_HPP
+#define MLPACK_CORE_CV_METRICS_ROCAUCSCORE_IMPL_HPP
+
+#include <mlpack/core/cv/metrics/accuracy.hpp>
+#include <iostream>
+
+namespace mlpack {
+namespace cv {
+
+template<size_t PC /* PositiveClass */>
+template<typename MLAlgorithm, typename DataType>
+double ROC_AUC<PC>::Evaluate(MLAlgorithm& model,
+                              const DataType& data,
+                              const arma::Row<size_t>& labels)
+{
+  util::CheckSameSizes(data, labels, "ROC_AUC::Evaluate()");
+
+  arma::mat probabilities;
+  model.Classify(data, probabilities);
+
+  // compute labels with "1" for positive class and "0" for the other
```

Comment:
```suggestion
  // Compute labels with "1" for positive class and "0" for the other.
```

See https://github.com/mlpack/mlpack/wiki/DesignGuidelines#comments for more details, the same applies to other comments as well.